client/galaxy/scripts/components/Panels/ToolBox.vue

```diff
@@ -11,6 +11,11 @@
         </div>
         <div class="unified-panel-controls">
             <tool-search :query="query" placeholder="search tools" @onQuery="onQuery" @onResults="onResults" />
+
+            <div class="float-none" v-if="results">
+                <button class="btn btn-secondary btn-sm" v-if="!show" @click="onToggle">Show Categories</button>
+                <button class="btn btn-secondary btn-sm" v-if="show" @click="onToggle">Hide Categories</button>
```

Comment:
We should use a computed variable for the button label, then a single `<button/>` component would be enough.